### PR TITLE
#168049996 On successful registration, users should receive a JWT

### PR DESCRIPTION
**Pull request by didasmbalanya**
#### What does this PR do?
 - Allow users to sign up
 - Create tests for register user functionality
 - Add swagger functionality for register new user

#### Description of Task to be completed?
 - Create a register user endpoint.
 - Return a JWT to the client machine on successful signup
 - Document feature using Swagger

#### How should this be manually tested?
 - After pulling from this branch
 - set up port environments as in env.example
 - Run docker-compose up --build
 - Endpoint: localhost:3000/api/v1/auth/register
 - On the request body, use firstname, lastname, username, email, and password as they are required fields.
 - you will get a response on success or fauilure
#### What are the relevant pivotal tracker stories?
[#168049996](https://www.pivotaltracker.com/story/show/168049996)
#### Screenshots (if appropriate)
Screen Shot 2019-09-10 at 21 21 47
Screen Shot 2019-09-10 at 21 30 40

 - On successful sign up the user receives a JWT.
 - A user can not sign up twice.
 - below is a screenshot of the UI mockup
Screen Shot 2019-09-11 at 10 27 49

#### Questions:

### CODE REVIEW

**Review by gik-hub — APPROVED** [review]
Good work, However, I have realized your UserService methods don't have a JSON doc, would you please consider adding it as well.

**Review by mucyomiller — APPROVED** [review]
LGTM

File: src/controllers/UserController.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  /**
+   * @param {object} [req] object
+   * @param {object} [res] object
+   * @returns {object} user on succesful sign up\q
+   * */
+  static async register(req, res) {
+    /**
+     * @returns user object on succesful sign up
+    */
+    const foundUser = await userService.findUserByEmail(req.body.email);
```
Comment (Paulvitalis200) [inline src/controllers/UserController.js:40]: Wrap line `40` to `49` within a `try catch` block to handle exceptions or errors.

**Comment by Paulvitalis200** [comment]
Add steps for testing with docker too.

File: src/controllers/UserController.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    if (foundUser) {
+      res.status(409).send({ status: 409, error: `User with email ${req.body.email} already exists` });
+    } else {
+      req.body.is_admin = false;
+      req.body.password = bcrypt.hashSync(req.body.password, 8);
+      const newUser = await userService.addUser(req.body);
+      const token = jwtSign(req.body);
+      res.status(201).send({ status: 201, data: getPublicProfile(newUser), token });
```
Comment (Paulvitalis200) [inline src/controllers/UserController.js]: I think we can also add a message like `Successful sign up`. Think of a good success message

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**